### ISSUE TITLE
Pelican client plugin errors

### ISSUE THREAD

**Issue by joereuss12** [PelicanPlatform/pelican#210]
From experimenting with the Pelican client on Submit1, there are errors occurring when using this with HTCondor as a custom file transfer plugin. The error always seems to be:
`012 (17023099.000.000) 2023-10-10 13:02:11 Job was held.
        Transfer input files failure at execution point slot1_157@e4008.chtc.wisc.edu while receiving files from access point submit-1. Details: file transfer plugin stash_plugin exited with code 1, unable to open output file /var/lib/condor/execute/slot1/dir_3696929/.stash_plugin.out
        Code 13 Subcode 256`

The submit file used is:
`executable              = job.sh`
`log                     = job.log`
`output                  = job.out`
`error                   = job.err`
`transfer_plugins        = osdf=stash_plugin`
`transfer_input_files    = osdf:///chtc/staging/jrreuss/foo-v6.txt`
`should_transfer_files   = Yes`
`when_to_transfer_output = ON_EXIT`
`request_cpus            = 1`
`request_memory          = 512M`
`request_disk            = 1G`
`queue`

I have also tried this with:
`transfer_plugins        = pelican=stash_plugin`
`transfer_input_files    = pelican://osg-htc.org/chtc/staging/jrreuss/foo-v6.txt`

Every time I run this I encounter this error and it will need investigating

**Comment by joereuss12** [comment]
Did substantial digging and still have no idea what might be causing this problem because I cannot get any amount of debug output to lead me to it. I discovered that over on the EP side of things, the .stash_plugin.out file is not being created at all which is why condor cannot open the file to begin with so I know it is not a permissions error. What is weird is that it seems like it is not exiting from `plugin.go` and instead somewhere else because no matter what I change it always states it exits with exit code 1. Below is everything I have tried some of which with the help of Cole:
- runing startd's PID through strace, could not find much info to help me
- Putting sleeps before/changing codes of the exit 1's in plugin.go. Still does exit(1) each time
- Tried putting a wrapper script around the plugin (made it sleep after running to find the .out file not created) trying to run it with the -d flag and capture the output. The debug output file always ends up blank
- Tried making the outfile earlier in the code

My current assumption is that maybe there is something wrong with how it is set up or somehow condor does not know what to do with stash_plugin causing it to throw an exit code 1 before anything is run or set up? Just very weird that the .stash_plugin.in file is created but the outfile is not. I'm not very sure if this is an issue on the condor side or Pelican side of things but leaning towards either Pelican or something I might be doing wrong.
